src/btree/bt_split.c

```diff
@@ -187,7 +187,7 @@ __split_safe_free(WT_SESSION_IMPL *session,
 		exclusive = true;
 
 	if (exclusive) {
-		__wt_free(session, p);
+		__wt_overwrite_and_free_len(session, p, s);
```

Comment:
Note to reviewers: this is intended to catch future bugs more reliably.  It matches what we do in `__wt_split_stash_discard`.